### PR TITLE
Pytest Style analysis/test_hole.py

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/analysis/test_hole.py [hunk 1]
```diff
@@ -21,27 +21,22 @@
...
-import mpl_toolkits.mplot3d
-
-
-import os
-import errno
+from six.moves import range
 
-from MDAnalysisTests.datafiles import PDB_HOLE, MULTIPDB_HOLE
-from MDAnalysisTests import (executable_not_found, module_not_found,
-                             tempdir, in_dir)
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_hole.py]: import order is  

1. python std library
1. external packages like numpy/pandas/...
1. mdanalysis specific packages
1. testing specific packages

File: testsuite/MDAnalysisTests/analysis/test_hole.py [hunk 2]
```diff
@@ -21,27 +21,22 @@
...
 import MDAnalysis.analysis.hole
+import matplotlib
+import mpl_toolkits.mplot3d
+import numpy as np
+import pytest
 from MDAnalysis.analysis.hole import HOLEtraj, HOLE
-
-from numpy.testing import (TestCase, dec,
-                           assert_equal, assert_almost_equal,
+from MDAnalysisTests import (executable_not_found, tempdir, in_dir)
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_hole.py]: please remove the tempdir dependency

File: testsuite/MDAnalysisTests/analysis/test_hole.py [hunk 3]
```diff
@@ -68,116 +63,95 @@ def setUp(self):
...
-            cls.H = None
+    @pytest.fixture()
+    def universe(self):
+        return MDAnalysis.Universe(self.filename)
 
-        cls.frames = [ts.frame
-                      for ts in cls.universe.trajectory[cls.start:cls.stop]]
+    @pytest.fixture()
+    def H(self, universe):
+        with tempdir.in_tempdir():
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_hole.py]: use a tmpdir_factory here

File: testsuite/MDAnalysisTests/analysis/test_hole.py [hunk 4]
```diff
@@ -55,10 +50,10 @@ def rlimits_missing():
     return False
 
 
+@pytest.mark.skipif(executable_not_found("hole"), reason="Test skipped because HOLE not found")
 class TestHOLE(TestCase):
     filename = PDB_HOLE
 
-    @dec.skipif(executable_not_found("hole"), msg="Test skipped because HOLE not found")
     def setUp(self):
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_hole.py]: should be a fixture

**Comment by utkbansal** [comment]
@kain88-de Anything else?

File: testsuite/MDAnalysisTests/analysis/test_hole.py [hunk 5]
```diff
@@ -55,129 +55,108 @@ def rlimits_missing():
...
 
-class TestHOLE(TestCase):
+@pytest.mark.skipif(executable_not_found("hole"), reason="Test skipped because HOLE not found")
+class TestHOLE(object):
     filename = PDB_HOLE
-
-    @dec.skipif(executable_not_found("hole"), msg="Test skipped because HOLE not found")
-    def setUp(self):
+    
+    @pytest.fixture()
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_hole.py]: Shouldn't this be a static method for pytest to work?

Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_hole.py]: I don't think so. I can move `filename = PDB_HOLE` to the fixture though and make it a staticmethod.

Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_hole.py]: so far you made all fixtures in a class static. I was wondering why this isn't necessary here.

Comment (utkbansal) [inline testsuite/MDAnalysisTests/analysis/test_hole.py]: @kain88-de I sometimes need to access `self` from within the fixture. That is when I don't make it static.

**Pull request merged**